### DISCUSSION TITLE
sprite animations get stuck in the first frame

### DISCUSSION THREAD

**Discussion by DarkiCreepKato** [HaxeFlixel/flixel#2847] [Q&A]
hello, i was trying to make a weird game for testing if i was able to code some quick game by myself and show it to my friends, but the animations for the text that i writed myself in paint tool sai and made a sprite in adobe animate just plays the first frame, i just was able to load two, and they both just play the first frame, i was trying to code that the first text shows a text animation, then when you press space, the text changes and plays another text animation, (i was going to make more texts) but umm it just plays the first frame, what did i maded wrong and how can i fix it?, at first it worked right but when i made that the first text dissapear and a more couple things that i dont remember, because the freaky text didnt even appear and then fix that, that happened, so, heres the code in playstate:

```hx
package;

import flixel.FlxG;
import flixel.FlxSprite;
import flixel.FlxState;
import flixel.input.FlxInput;
import flixel.util.FlxTimer;

class PlayState extends FlxState
{
	var fondo_negro:FlxSprite;
	var poyo_talk:FlxSprite;
	var burbuja_texto:FlxSprite;
	var textopoyo1:FlxSprite;
	var textopoyo2:FlxSprite;

	override public function create():Void
	{
		super.create();
		FlxG.sound.playMusic(AssetPaths.Keepin_up_with_the_kids_msi__ogg, 1, true);

		fondo_negro = new FlxSprite();
		fondo_negro.loadGraphic(AssetPaths.fondo_negro__png);
		fondo_negro.x = 0;
		fondo_negro.y = 0;
		add(fondo_negro);

		burbuja_texto = new FlxSprite();
		burbuja_texto.loadGraphic(AssetPaths.burbuja_de_texto__png);
		burbuja_texto.x = 0;
		burbuja_texto.y = 0;
		new FlxTimer().start(5.0, function(t:FlxTimer)
		{
			add(burbuja_texto);
		});

		textopoyo1 = new FlxSprite();
		textopoyo1.loadGraphic(AssetPaths.texto_poyo_2__png, true, 536, 308);
		textopoyo1.animation.add("texto_poyo", [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14], 14, false);
		textopoyo1.x = 0;
		textopoyo1.y = 0;
		new FlxTimer().start(5.0, function(t:FlxTimer)
		{
			add(textopoyo1);

			textopoyo1.animation.play("texto_poyo");
			textopoyo1.animation.play("texto_poyo");
		});

		textopoyo2 = new FlxSprite();
		textopoyo2.loadGraphic(AssetPaths.texto_poyo_3__png, true, 536, 308);
		textopoyo2.animation.add("texto_poyo", [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,], 14, false);
		textopoyo2.x = 0;
		textopoyo2.y = 0;

		poyo_talk = new FlxSprite();
		poyo_talk.loadGraphic(AssetPaths.poyo_talk__png, true, 545, 309);
		poyo_talk.animation.add("poyotalk0", [0], 14, false);
		poyo_talk.animation.add("poyotalk1", [1], 14, false);
		poyo_talk.animation.add("poyotalk2", [2], 14, false);
		poyo_talk.animation.add("poyotalk3", [3], 14, false);
		poyo_talk.animation.add("poyotalk4", [4], 14, false);
		poyo_talk.animation.add("poyotalk5", [5], 14, false);
		poyo_talk.animation.add("poyotalk6", [6], 14, false);
		poyo_talk.animation.add("poyotalk7", [6], 14, false);
		poyo_talk.x = 0;
		poyo_talk.y = 0;
		new FlxTimer().start(5.0, function(t:FlxTimer)
		{
			poyo_talk.animation.play("poyotalk0");
			add(poyo_talk);
			poyo_talk.animation.play("poyotalk0");
		});
	}

	override public function update(elapsed:Float)
	{
		if (FlxG.keys.pressed.SPACE)
			poyo_talk.animation.play("poyotalk1");
		if (FlxG.keys.pressed.SPACE)
			textopoyo1.visible = false;
		if (FlxG.keys.pressed.SPACE)
			add(textopoyo2);
		if (FlxG.keys.pressed.SPACE)
			textopoyo2.animation.play("texto_poyo");
		textopoyo2.animation.play("texto_poyo");
	}
}
```

**Comment by Geokureli** [comment]
be sure to trim your examples down to it's simplest form so we can easily recreate it locally. does this happen with every sprite or just a specific?  does this example need the music and the timers? there's a lot of odd things you're doing here, in general so it's hard to diagnose the issue, and these timers might just add confusion. Because you're never adding `textopoyo2` and everything other than `fondo_negro` is added after 5 seconds, for no real reason

Can you list the dimensions of every relevant image?

**Reply by Geokureli** [reply]
I tried poyo_talk, by itself, for "poyo_talk__png" i just used a 3815x309 image with numbers on each frame. Everything worked as expected, maybe the frames in your sprite sheet just look like the first frame?

```hx
package;

import flixel.FlxG;
import flixel.FlxSprite;

class AnimationFrameTestState2847 extends flixel.FlxState
{
	var poyo_talk:FlxSprite;

	override public function create():Void
	{
		super.create();

		poyo_talk = new FlxSprite();
		poyo_talk.loadGraphic(AssetPaths.poyo_talk__png, true, 545, 309);
		poyo_talk.animation.add("poyotalk0", [0], 14, false);
		poyo_talk.animation.add("poyotalk1", [1], 14, false);
		poyo_talk.animation.add("poyotalk2", [2], 14, false);
		poyo_talk.animation.add("poyotalk3", [3], 14, false);
		poyo_talk.animation.add("poyotalk4", [4], 14, false);
		poyo_talk.animation.add("poyotalk5", [5], 14, false);
		poyo_talk.animation.add("poyotalk6", [6], 14, false);
		poyo_talk.animation.add("poyotalk7", [6], 14, false); // <-- is 6 a typo, should be 7?
		add(poyo_talk);
		poyo_talk.animation.play("poyotalk0");
	}

	override public function update(elapsed:Float)
	{
		if (FlxG.keys.pressed.SPACE)
			poyo_talk.animation.play("poyotalk1");
	}
}
```

btw, you don't need to make animations for each frame, you can just set `poyo_talk.animation.frameIndex = 1;` like so:

```hx
poyo_talk = new FlxSprite();
poyo_talk.loadGraphic(AssetPaths.poyo_talk__png, true, 545, 309);
add(poyo_talk);
poyo_talk.animation.frameIndex = 0;
```

**Reply by DarkiCreepKato** [reply]
i think it doesnt needs the timers and music um ok here is the dimensions of the images texto poyo 2: 2048 x 2048 texto poyo 3: 2048 x 2048
poyo talk: 2048 x 1024 fondo negro: 1024 x 512 burbuja de texto: 1024 x 512

**Reply by DarkiCreepKato** [reply]
oh okay

**Reply by Geokureli** [reply]
why is poyo_talk 2048x1024 but the frame size is 545x309? the image size should be a multiple of the frame size

**Comment by DarkiCreepKato** [comment]
idk but it works so, here is the image that just showsthe first frame 
![image](https://github.com/HaxeFlixel/flixel/assets/136510357/87ad153b-6bd2-4008-bf34-35f8afd0bd09)

**Reply by DarkiCreepKato** [reply]
![texto poyo 2](https://github.com/HaxeFlixel/flixel/assets/136510357/5348cb66-d4f0-43c0-9bcf-168a1e705a70)

**Comment by DarkiCreepKato** [comment]
it just happens to the texto poyo sprites

**Comment by DarkiCreepKato** [comment]
translation poyo text sprite

**Comment by DarkiCreepKato** [comment]
![texto poyo 3](https://github.com/HaxeFlixel/flixel/assets/136510357/b45cb3df-0bae-4182-bb5c-efcca64ca2be)
here is the other that just shows the first frame

**Reply by Geokureli** [reply]
you should look into [FlxTypeText](https://api.haxeflixel.com/flixel/addons/text/FlxTypeText.html)

**Reply by DarkiCreepKato** [reply]
oh okay I'll try it

**Reply by DarkiCreepKato** [reply]
i dont understand sjsjjdkdj 😔
